### PR TITLE
Loading view

### PR DESCRIPTION
**Pull request by davideme**
Adding basic super of Loading UI with default implementation and UIView customizable.

### CODE REVIEW

File: BothamUI/BothamCustomView.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    func xibSetup() {
+        view = loadViewFromNib()
+        view.frame = bounds
+        view.autoresizingMask = [UIViewAutoresizing.FlexibleWidth, UIViewAutoresizing.FlexibleHeight]
+        addSubview(view)
+    }
+
+    func loadViewFromNib() -> UIView {
+        return NSBundle(forClass: self.dynamicType).loadNib(name: String(self.dynamicType), owner: self)
+    }
```
Comment (pedrovgs) [inline BothamUI/BothamCustomView.swift:33]: Shouldn't these methods be private?

**Comment by pedrovgs** [comment]
Should we rename the the group were we are placing some extensions like the one for `NSBundle` from `Library` to `Extensions`?

File: BothamUI/BothamLoadingUI.swift [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+//
+//  Created by Davide Mendolia on 24/11/15.
+//  Copyright © 2015 GoKarumi S.L. All rights reserved.
+//
+
+import Foundation
+
+public protocol BothamLoadingUI: BothamUI {
+    func showLoader()
+    func hideLoader()
```
Comment (pedrovgs) [inline BothamUI/BothamLoadingUI.swift:13]: Don't you need to declare these methods as public like you are already doing with the protocol?

Comment (davideme) [inline BothamUI/BothamLoadingUI.swift:13]: Visibity modifier are not allowed in protocol. everything has always the same visibilty of the protocol.

Comment (pedrovgs) [inline BothamUI/BothamLoadingUI.swift:13]: Cool thanks!

File: Example/Example/ComicsPresenter.swift [hunk 3]
```diff
@@ -10,16 +10,21 @@ import Foundation
 import BothamUI
 
 class ComicsPresenter: BothamPresenter {
-    weak var ui: ComicsUI?
+    var ui: ComicsUI?
```
Comment (pedrovgs) [inline Example/Example/ComicsPresenter.swift]: Can't we change the this attribute declaration to non optional?

Comment (davideme) [inline Example/Example/ComicsPresenter.swift]: My bad that should be weak. It's was only there for testing.

**Comment by pedrovgs** [comment]
We've been discussing about the testing approach and the solution is not straightforward so I guess we can review the test we have failing and we can merge this PR :+1: :)

**Comment by davideme** [comment]
@pedrovgs I added Presenter test and we will continue our investigation with ViewController test in parallel

**Comment by pedrovgs** [comment]
Let's go to merge this PR :) :+1:

**Pull request merged**